File: Ciklum Kyiv 2014/MvxMadness/MvxMadness/FirstViewModel.fs
namespace MvxMadness

open System

open Cirrious.CrossCore.IoC
open Cirrious.MvvmCross.ViewModels

type internal ComplexLogic() =
    member this.Reverse (s:string) = new string(s |> Seq.toArray |> Array.rev)

type FirstViewModel() = 
    inherit MvxViewModel()

    let mutable input : string = "F# Madness"
    let mutable output : string = ""
    let logic = new ComplexLogic()

    member this.Input
        with get () = input
        and set (value) = 
            input <- value 
            this.RaisePropertyChanged "Input"

    member this.Output
        with get () = output
        and set (value) =         
            output <- value 
            this.RaisePropertyChanged "Output"
    
    member private this.DoMagic () =
            this.Output <- logic.Reverse(this.Input)
    
    member this.Magic
        with get () =
            new MvxCommand (fun () -> this.DoMagic())

File: Ciklum Kyiv 2014/MvxSample/MvxSample.Core/FirstViewModel.cs
using System.Windows.Input;
using Cirrious.MvvmCross.ViewModels;

namespace MvxSample.Core
{

    public class FirstViewModel 
        : MvxViewModel
    {
        string _input = "Xamarin and MvvmCross";

        public string Input {
            get { return _input; }
            set {
                _input = value;
                RaisePropertyChanged (() => Input);
            }
        }

        string _output;

        public string Output {
            get { return _output; }
            set {
                _output = value;
                RaisePropertyChanged (() => Output);
            }
        }

        ICommand _magic;

        public ICommand Magic {
            get {
                _magic = _magic ?? new MvxCommand (DoMagic);
                return _magic;
            }
        }

        void DoMagic ()
        {
            var logic = new ComplexLogic ();
            Output = logic.Reverse (Input);
        }
    }
    
}